pkg/spiffe/spiffe.go

```diff
@@ -64,12 +73,49 @@ func SetTrustDomain(value string) {
 	trustDomainMutex.Unlock()
 }
 
+// TODO(incfly): rename this and SetTrustDomain as GetLocalTrustDomain for clarity.
 func GetTrustDomain() string {
 	trustDomainMutex.RLock()
 	defer trustDomainMutex.RUnlock()
 	return trustDomain
 }
 
+func GetTrustDomainByCluster(clusterID string) string {
+	i, ok := trustDomainInfoMap.Load(clusterID)
+	if ok {
+		return i.(ClusterTrustDomainInfo).TrustDomain
+	}
+	return GetTrustDomain()
+}
+
+func SetTrustDomainByCluster(clusterID string, trustDomain string) {
+	if trustDomain == "" || clusterID == "" {
+		spiffeLog.Errorf("Trust Domain and cluster ID can't be empty, cluster %v, trust domain %v",
+			clusterID, trustDomain)
+		return
+	}
+	trustDomainInfoMap.Store(clusterID, ClusterTrustDomainInfo{
+		ClusterID:   clusterID,
+		TrustDomain: trustDomain,
+	})
+}
+
+// DumpDebugInfo returns the current information related to spiffe.
+// TODO(incfly): change to json format before merge.
+func DumpDebugInfo() string {
+	var b bytes.Buffer
+	b.WriteString(fmt.Sprintf("Local trust domain: %v\n", GetTrustDomain()))
```

Comment:
the spiffez returns application/json but this is not json

Reply:
And the /debug interface is getting deprecated, plus all this should be in mesh config, not yet another ad-hoc config based on globals.

Reply:
obsolete removed.